examples/bridge-app/linux/main.cpp

```diff
@@ -358,6 +402,33 @@ void HandleDeviceSwitchStatusChanged(DeviceSwitch * dev, DeviceSwitch::Changed_t
     }
 }
 
+void HandleDevicePowerSourceStatusChanged(DevicePowerSource * dev, DevicePowerSource::Changed_t itemChangedMask)
+{
+    if (itemChangedMask &
+        (DevicePowerSource::kChanged_Reachable | DevicePowerSource::kChanged_Name | DevicePowerSource::kChanged_Location))
+    {
+        HandleDeviceStatusChanged(static_cast<Device *>(dev), (Device::Changed_t) itemChangedMask);
+    }
+
+    if (itemChangedMask & DevicePowerSource::kChanged_BatLevel)
+    {
+        uint8_t batChargeLevel = dev->GetBatChargeLevel();
+        MatterReportingAttributeChangeCallback(dev->GetEndpointId(), ZCL_POWER_SOURCE_CLUSTER_ID,
+                                               ZCL_POWER_SOURCE_BAT_CHARGE_LEVEL_ATTRIBUTE_ID, CLUSTER_MASK_SERVER,
+                                               ZCL_INT8U_ATTRIBUTE_TYPE, &batChargeLevel);
+    }
+
+    if (itemChangedMask & Device::kChanged_Name)
+    {
+        uint8_t zclName[kPowerSourceDescriptionSize];
+        MutableByteSpan zclNameSpan(zclName);
+        MakeZclCharString(zclNameSpan, dev->GetDescription().c_str());
+        MatterReportingAttributeChangeCallback(dev->GetEndpointId(), ZCL_POWER_SOURCE_CLUSTER_ID,
+                                               ZCL_POWER_SOURCE_DESCRIPTION_ATTRIBUTE_ID, CLUSTER_MASK_SERVER,
+                                               ZCL_CHAR_STRING_ATTRIBUTE_TYPE, zclNameSpan.data());
```

Comment:
This can't possibly be useful to the callee: they have no idea how much data is there.  Just call the three-arg version without the data value?

Reply:
Thanks, I'll just  use three-arg version.